paparazzi/src/test/java/app/cash/paparazzi/internal/apng/APNGWriterTest.kt

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.cash.paparazzi.internal.apng
+
+import app.cash.paparazzi.accessibility.RenderSettings.toColorInt
+import app.cash.paparazzi.internal.apng.PngConsts.Header
+import app.cash.paparazzi.internal.apng.PngConsts.Header.ACTL
+import app.cash.paparazzi.internal.apng.PngConsts.Header.FCTL
+import app.cash.paparazzi.internal.apng.PngConsts.Header.FDAT
+import app.cash.paparazzi.internal.apng.PngConsts.Header.IDAT
+import app.cash.paparazzi.internal.apng.PngConsts.Header.IHDR
+import app.cash.paparazzi.internal.apng.PngConsts.PNG_BITS_PER_PIXEL
+import app.cash.paparazzi.internal.apng.PngConsts.PNG_COLOR_TYPE_RGBA
+import app.cash.paparazzi.internal.apng.PngConsts.PNG_SIG
+import com.google.common.truth.Truth.assertThat
+import okio.Buffer
+import okio.BufferedSource
+import okio.FileSystem
+import okio.Path.Companion.toPath
+import okio.buffer
+import okio.internal.commonToUtf8String
+import org.junit.Test
+import java.awt.Color
+import java.awt.Point
+import java.awt.image.BufferedImage
+import java.io.File
+import java.util.zip.CRC32
+import java.util.zip.Inflater
+
+class APNGWriterTest {
+
+  @Test
+  fun writesAnimationMetadata() {
+    val testFile = File.createTempFile("writesAnimationMetadata.png", null)
```

Comment:
will this temp file be cleaned up?  Should a TemporaryFolder rule be used instead?